src/index.ts

```diff
@@ -217,6 +240,21 @@ export class Tado {
     return response.data as R;
   }
 
+  /**
+   * Makes an API call to the provided TadoX URL with the specified method and data.
+   *
+   * @typeParam R - The type of the response
+   * @typeParam T - The type of the request body
+   * @param url - The endpoint to which the request is sent. If the URL contains "https", it will be used as is.
+   * @param method - The HTTP method to use for the request (e.g., "get", "post").
+   * @param data - The payload to send with the request, if applicable.
+   * @returns A promise that resolves to the response data.
+   */
+  async apiCallX<R, T = unknown>(url: string, method: Method = "get", data?: T): Promise<R> {
```

Comment:
Can't we move this logic into apiCall itself? Something like 

```typescript
callUrl = <isX> ? tado_x_url : url;
```

Since home_id is not passed in if you follow my other comment, will have to add isX to parameter.

Reply:
I kept it separate to allow the use of v2 calls for the endpoints that haven't migrated but honestly I have no idea if they'll work with a tadoX home.

Reply:
Is there a tester in the community?

Reply:
A couple have volunteered from the NodeRed community and I've linked the PR from that codebase. Hopefully one of them can test. I plan to leave this unmerged until someone with a unit can check it.

Reply:
They could just add you to their homes :smile:

Reply:
Hmm, had a quick look at the apk and the only endpoint I can see used by the hops api service is

```
@GET("/homes/{homeId}/features")
```

I can't find `/roomsAndDevices` or `/rooms`. 

I can check a more recent version soon. But this could mean that v2 endpoints still work for X line.

Reply:
Interesting. That's an endpoint that isn't in the linked python repo.

Reply:
I am not sure hops is what we are thinking it is.

Reply:
I can confirm the new apk versions do not use these endpoints either. Best way to confirm might be to use the web app and tracking xhr requests when logged in to a home using X hardware.